libs/langgraph/src/prebuilt/react_agent_executor.ts

```diff
@@ -457,7 +457,7 @@ type WithStateGraphNodes<K extends string, Graph> = Graph extends StateGraph<
 
 const PreHookAnnotation = Annotation.Root({
   llmInputMessages: Annotation<BaseMessage[], Messages>({
-    reducer: messagesStateReducer,
+    reducer: (_, update) => messagesStateReducer([], update),
```

Comment:
Why don‘t you return it directly, but use that function again? Does it have any special function?

Reply:
We want to make sure that (1) we're sending the same messages state-wise (assigning IDs if they are not present) and (2) we still accept `Messages` as update instead of just `BaseMessage[]`